tests/test_divide_dataset.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+from pathlib import Path
+import json
+from scripts.divide_dataset import main
+DATASET = [
+            {
+                "company_name": "SSAB",
+                "SNI": "123456789",
+                "text": "Lorem Lipsum",
+                "phone_number": "123456789",
+                "org_number": "123456789",
+                "address": "123456789",
+                "municipality": "Stockgolm",
+                "postal_code": "123456789",
+                "url": "https://ssab.se/"
+            },
+            {
+                "company_name": "LKAB",
+                "SNI": "123456789",
+                "text": "Lorem Lipsum",
+                "phone_number": "123456789",
+                "org_number": "123456789",
+                "address": "123456789",
+                "municipality": "Stockgolm",
+                "postal_code": "123456789",
+                "url": "https://lkab.com/"
+            }
+        ]
+
+@pytest.fixture
+def temp_dataset(tmp_path):
+    """
+    Create a temporary input file 
+    The file has the same structure as the dataset.
+    """
+    input_file = tmp_path / "input_data.json"
+    with open(Path(input_file), 'w', encoding='utf-8') as f:
```

Comment:
Maybe make sure tmp_path exists before trying to open the file, so that the tests won't fail because of it:
```
Path(tmp_path).mkdir(parents=True, exist_ok=True)
```

Reply:
No need since the pytest fixture creates the temporary directory for us.